www/src/components/ui/hooks/use-copy-message.ts

```diff
@@ -0,0 +1,21 @@
+import { useCallback } from "react";
+
+import { useToast } from "./use-toast";
+
+export const useCopyMessage = () => {
+  const { toast } = useToast();
+
+  const handleCopyMessage = useCallback(
+    (textToCopy: string) => {
+      navigator.clipboard.writeText(textToCopy).then(() => {
+        toast({
+          title: "Copied to clipboard",
+          description: "The message has been copied to your clipboard.",
+        });
+      });
```

Comment:
`#would` use async/await, but curious to understand if you have a preference for using callbacks

Reply:
Tbh If I'm not mistaken, this function was already like that, then I've just copy and paste to this hook 😬 , but it's fixed and with error handling now!